package.json

```diff
@@ -33,6 +33,7 @@
     "axios": "^0.19.2",
     "console-testing-library": "^0.3.1",
     "eslint-config-react-app": "^5.0.1",
+    "immutable": "^4.0.0-rc.12",
```

Comment:
I pulled in a new dev dependency to use its type. Let me know if you'd rather not add one more dependency and maybe we can just use a type of `any` or `unknown` above or something, although it'll be less precise.

Reply:
Having this just as a dev dependency will make bundlers like yarn2 with pnp complain that a dependency is missing. Correct would probably be an optional peer dependency.

Also, this might be just imported as a type directly by using 
```diff
-immutable?: typeof Immutable
+immutable?: import('immutable')
```
down there.
That way it will never be a runtime require - which might prevent tree-shaking.

Reply:
Sounds good. I'll add it as an optional dependency (couldn't find anything specific to optional *dev* dependency).

Also `immutable?: import('immutable')` did not work so I changed it to `immutable?: typeof import('immutable')`, otherwise TS complains:
![Screen Shot 2020-10-06 at 1 18 02 PM](https://user-images.githubusercontent.com/1617767/95255745-b53cf980-07d6-11eb-9f4e-311f36e06e7d.png)

Reply:
Optional *peer* dependency - it's not a dev dependency, because RTK's dev dependencies are only installed when developing against RTK. But you want it to point against the right package if RTK and immutable are installed while *using* RTK.

So it should be a "real" dependency, but the loosest form of that.
At least that's how I understood the usage of that - @markerikson might know more though.

See https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md - it's fairly new but should be supported by the bigger bundlers by now.

Reply:
Oh interesting. I had never heard of that. I'll take a look.

Reply:
Now that it's a peer dependency, TS complains because it's not installed locally. Do you know how to fix that part properly?
![Screen Shot 2020-10-06 at 2 35 25 PM](https://user-images.githubusercontent.com/1617767/95262553-34372f80-07e1-11eb-889c-729d22be95d9.png)